module/purpose/PS3Walk/src/PS3Walk.cpp

```diff
@@ -72,62 +83,146 @@ namespace module::purpose {
                         case AXIS_LEFT_JOYSTICK_HORIZONTAL:
                             // y is left relative to robot
                             // strafe[1] = -event.value;
-                            rotationalSpeed = static_cast<float>(-event.value);
+                            walk_command.z() = static_cast<double>(-event.value) / std::numeric_limits<short>::max()
+                                               * cfg.maximum_rotational_velocity;
                             break;
                         case AXIS_LEFT_JOYSTICK_VERTICAL:
                             // x is forward relative to robot
-                            strafe[0] = -event.value;
+                            walk_command.x() = static_cast<double>(-event.value) / std::numeric_limits<short>::max()
+                                               * cfg.maximum_forward_velocity;
+                            break;
+                        case AXIS_RIGHT_JOYSTICK_VERTICAL:
+                            head_pitch = static_cast<double>(-event.value) / 32767.0;
+                            break;
+                        case AXIS_RIGHT_JOYSTICK_HORIZONTAL:
+                            head_yaw = static_cast<double>(-event.value) / 32767.0;
                             break;
-                        case AXIS_RIGHT_JOYSTICK_VERTICAL: headPitch = static_cast<float>(-event.value); break;
-                        case AXIS_RIGHT_JOYSTICK_HORIZONTAL: headYaw = static_cast<float>(-event.value); break;
                     }
                 }
                 else if (event.isButton()) {
                     // event was a button event
                     switch (event.number) {
-                        case BUTTON_TRIANGLE:
+                        case BUTTON_START:
                             if (event.value > 0) {  // button down
                                 if (moving) {
-                                    NUClear::log("Stop walking");
-                                    emit<Task>(std::make_unique<Walk>(Eigen::Vector3d::Zero()), 1);
+                                    log<NUClear::INFO>("Stop walking");
+                                    emit<Task>(std::make_unique<Walk>(Eigen::Vector3d::Zero()), 2);
+                                    emit<Task>(std::unique_ptr<Walk>(nullptr));
```

Comment:
These are conflicting. The Director won't, run them one at a time, it'll run them together as a task pack at the end of the reaction. I'd have to look at the order of the code but I assume the first line would just do nothing. Do you have any issues with the walk freezing when stopped (rather than cleanly finishing)?
I wouldn't bother with the `nullptr`. Did you have issues without it?